src/components/reference-editor/index.js

```diff
@@ -0,0 +1,54 @@
+import React, { PropTypes } from 'react';
+import reactStamp from 'react-stamp';
+
+import EditorFactory from 'components/editor';
+
+import createMentionPlugin, { defaultSuggestionsFilter } from 'draft-js-mention-plugin';
+import createUndoPlugin from 'draft-js-undo-plugin';
+import createLinkifyPlugin from 'draft-js-linkify-plugin';
+
+import 'draft-js-mention-plugin/lib/plugin.css';
+import 'draft-js-undo-plugin/lib/plugin.css';
+import 'draft-js-linkify-plugin/lib/plugin.css';
+
+import MentionComponent from './mention-text';
+import SuggestionsFactory from './suggestions';
+
+export default function ({
```

Comment:
I am a little concerned with this approach. This makes the reference editor export a "factory factory" (i.e. a factory that returns a factory that returns a react component). And we must have all invocations throughout the application call a function that, as far as the calling code is concerned, doesn't do anything, just to make use of the component. And that's all just to make testing a little easier.

Can we approach this without the factory factory?